etcd: Increase status check timeout to 10 seconds

tgraf opened the pull request:
If the node is under heavy CPU load, the status check may take longer
than 5 seconds. Increase it to 10 seconds and also make it configurable
with an environment variable.

ungureanuvladvictor commented on pkg/kvstore/etcd.go:
```diff
@@ -211,6 +211,13 @@ func (e *etcdModule) newClient(ctx context.Context, opts *ExtraOptions) (Backend
 	// register etcd module for use
 	registerBackend(EtcdBackendName, etcdInstance)
+
+	if duration := os.Getenv("CILIUM_ETCD_STATUS_CHECK_INTERVAL"); duration != "" {
```
@tgraf any reason not to have this as a CLI option as well?

tgraf replied:
Yes. This option is very unlikely to be changed frequently and a full-blown CLI option makes backporting harder and also pollutes the `--help` output massively. We already have way too many options.

The environment variable is the last resort for advanced users

ungureanuvladvictor replied:
Got it, makes sense.

```
We already have way too many options.
``` 

+1 to the above.

tgraf committed:
etcd: Increase status check timeout to 10 seconds

If the node is under heavy CPU load, the status check may take longer
than 5 seconds. Increase it to 10 seconds and also make it configurable
with an environment variable.

Signed-off-by: Thomas Graf <thomas@cilium.io>

pchaigno reviewed (approved):
LGTM!
